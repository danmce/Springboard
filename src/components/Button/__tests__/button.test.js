import React from 'react';
import * as enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '../Button';
import UxPinWrapper from '../../../wrappers/UXPinWrapper';

test('renders error Button correctly', () => {
  const button = (
    <UxPinWrapper>
      <Button variant="primary">Test</Button>
    </UxPinWrapper>
  );

  expect(toJson(enzyme.mount(button))).toMatchSnapshot('enzyme.mount');
});
