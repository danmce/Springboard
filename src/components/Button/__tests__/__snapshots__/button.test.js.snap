// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error Button correctly: enzyme.mount 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: #008FC9;
  border: 0;
  border-radius: 4px;
  background-color: #008FC9;
}

<UXPinWrapper>
  <ThemeProvider
    theme={
      Object {
        "buttons": Object {
          "primary": Object {
            "backgroundColor": "primary",
          },
          "secondary": Object {
            "backgroundColor": "palette.white",
            "borderColor": "primary",
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "primary",
          },
        },
        "colors": Object {
          "palette": Object {
            "black": "#000000",
            "blue_10": "#DFF5FF",
            "blue_100": "#00486B",
            "blue_20": "#ADE5FF",
            "blue_30": "#74D4FD",
            "blue_40": "#34C0F4",
            "blue_50": "#00ABE6",
            "blue_60": "#0095D5",
            "blue_70": "#008FC9",
            "blue_80": "#006DA4",
            "blue_90": "#005A87",
            "green_10": "#E8FFEF",
            "green_100": "#005315",
            "green_20": "#B8FCCD",
            "green_30": "#7BF5A6",
            "green_40": "#39E984",
            "green_50": "#00D761",
            "green_60": "#00C548",
            "green_70": "#34AC57",
            "green_80": "#009324",
            "green_90": "#00761D",
            "n_10": "#F9FBFD",
            "n_100": "#4b7185",
            "n_20": "#F1F4F8",
            "n_30": "#E9EEF2",
            "n_300": "#2C546C",
            "n_40": "#E1E8EC",
            "n_50": "#D7DFE2",
            "n_500": "#17384B",
            "n_60": "#BECBD1",
            "n_600": "#0E2936",
            "n_70": "#A6B7C0",
            "n_80": "#8ea4af",
            "n_90": "#77919D",
            "red_10": "#FFEAE9",
            "red_100": "#871917",
            "red_20": "#FFD3D3",
            "red_30": "#FFB5B4",
            "red_40": "#FF908F",
            "red_50": "#FF706E",
            "red_60": "#FF5452",
            "red_70": "#FF3936",
            "red_80": "#E11B16",
            "red_90": "#B31612",
            "white": "#FFFFFF",
            "yellow_10": "#FFF7E8",
            "yellow_100": "#C88600",
            "yellow_20": "#FFE6AA",
            "yellow_30": "#FFD877",
            "yellow_40": "#FFCD49",
            "yellow_50": "#FFC200",
            "yellow_60": "#FFB900",
            "yellow_70": "#FFB200",
            "yellow_80": "#FFA900",
            "yellow_90": "#E59600",
          },
          "primary": "#008FC9",
        },
      }
    }
  >
    <Button
      variant="primary"
    >
      <ForwardRef
        variant="primary"
      >
        <Styled(div)
          __css={
            Object {
              "appearance": "none",
              "bg": "primary",
              "border": 0,
              "borderRadius": 4,
              "color": "white",
              "display": "inline-block",
              "fontSize": "inherit",
              "lineHeight": "inherit",
              "px": 3,
              "py": 2,
              "textAlign": "center",
              "textDecoration": "none",
            }
          }
          as="button"
          tx="buttons"
          variant="primary"
        >
          <button
            className="emotion-0"
          >
            Test
          </button>
        </Styled(div)>
      </ForwardRef>
    </Button>
  </ThemeProvider>
</UXPinWrapper>
`;
